### DISCUSSION TITLE
React beautiful dnd Problem

### DISCUSSION THREAD

**Discussion by Melih4irat** [neuefische/questions#417] [Web] [answered]
Branch: https://github.com/Melih4irat/Capstone/tree/testdnd
verzeichnis: http://localhost:3000/dragtes
es geht sich um folgende Komponenten
Folder: "Dragtes"
Files: Kanban.js, TaskCard.js

Fehler in der Console: 

- react-beautiful-dndA setup problem was encountered.> Invariant failed: Draggable[id: 0]: Unable to find drag handle
- react-beautiful-dndUnable to find any drag handles in the context "1"

**Comment by actyralf** [comment, marked as answer]
This seems to be a problem with server side rendering and `react-beautiful-dnd`. To fix these issue, please try to follow the steps documented here under "Step 4: Run Your Next.js Application and Issues Occur" : https://levelup.gitconnected.com/using-react-beautiful-dnd-with-next-js-and-typescript-58a198b7266b
Btw: When you touch your `_document.js` file you could also consider fixing styled components "flicker" issues which are also caused by server side rendering. You can find the documentation here: https://github.com/neuefische/ffm-web-22-1/blob/main/docs/next-js-styled-components-ssr-fix.md

**Reply by actyralf** [reply]
I tried it, you have to follow both steps described there:

- In your `next.config.js` change the value for `reactStrictMode` to `false`
- Replace your `/pages/_document.js` with the following code:
```js
import Document, {Html, Head, Main, NextScript} from "next/document";
import {ServerStyleSheet} from "styled-components";
import {resetServerContext} from "react-beautiful-dnd";

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: App => props => sheet.collectStyles(<App {...props} />),
          enhanceComponent: Component => Component,
        });

      const initialProps = await Document.getInitialProps(ctx);
      resetServerContext();
      return {
        ...initialProps,
        styles: [initialProps.styles, sheet.getStyleElement()],
      };
    } finally {
      sheet.seal();
    }
  }

  render() {
    return (
      <Html>
        <Head>
          <link
            rel="icon"
            href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>🐙</text></svg>"
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```
